service/controller/v12/resource/instance/create.go

```diff
@@ -256,17 +307,20 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 
 		if !masterUpgradeInProgress && !workerUpgradeInProgess {
 			r.logger.LogCtx(ctx, "level", "debug", "message", "neither masters nor workers upgraded")
-			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("removing resource status '%s/%s'", Stage, InstancesUpgrading))
 
-			err := r.deleteResourceStatus(customObject, Stage, InstancesUpgrading)
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting resource status to '%s/%s'", Stage, DeploymentCompleted))
+
+			err := r.setResourceStatus(customObject, Stage, DeploymentCompleted)
 			if err != nil {
 				return microerror.Mask(err)
 			}
 
-			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("removed resource status '%s/%s'", Stage, InstancesUpgrading))
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("set resource status to '%s/%s'", Stage, DeploymentCompleted))
+
 			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-			return nil
 		}
+
+		return nil
```

Comment:
```suggestion
```
I think this is not needed. The next line seems to close this scope and after that is `return nil`.

Reply:
right